core/src/mindustry/logic/LExecutor.java

```diff
@@ -260,6 +261,19 @@ public void run(LExecutor exec){
                             res = b == null ? null : b.tile;
                             build = true;
                         }
+                        case block -> {
+                            if(exec.obj(block) instanceof Block) {
+                                if (exec.bool(enemy)) {
+                                    Building b = Units.findEnemyTile(unit.team, unit.x, unit.y, Float.MAX_VALUE, e -> !e.dead() && e.block == exec.obj(block));
```

Comment:
This is very inefficient and not an acceptable solution. You can't scan the whole map every time you want to fetch a block.

Reply:
Then what should i do?

Reply:
Not sure what the best solution is. Could try a loop through all building entities, or a cache that tracks buildings by type.

Reply:
hmmm ill see ok